### ISSUE TITLE
CI fails due to incompatible update of `mpmath`

### ISSUE THREAD

**Issue by kenya-sk** [quantumlib/Cirq#6475]
**Description of the issue**
CI pipeline fails due to an incompatible update in the mpmath library used by the Sympy library. 
The `cirq_pre_release` installs the pre-release version of `mpmath=="1.4.0a0"`.  
`mpmath.rational` is deorecated due to the update from v1.3.

**How to reproduce the issue**

```

docker build --target cirq_pre_release -t cirq_image_pre .
docker run cirq_image_pre python -c "import cirq; assert cirq.__version__ is not None"

```

<details>

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/cirq/__init__.py", line 19, in <module>
    from cirq._compat import __cirq_debug__, with_debug
  File "/usr/local/lib/python3.9/site-packages/cirq/_compat.py", line 32, in <module>
    import sympy
  File "/usr/local/lib/python3.9/site-packages/sympy/__init__.py", line 30, in <module>
    from sympy.core.cache import lazy_function
  File "/usr/local/lib/python3.9/site-packages/sympy/core/__init__.py", line 9, in <module>
    from .expr import Expr, AtomicExpr, UnevaluatedExpr
  File "/usr/local/lib/python3.9/site-packages/sympy/core/expr.py", line 4159, in <module>
    from .mul import Mul
  File "/usr/local/lib/python3.9/site-packages/sympy/core/mul.py", line 2193, in <module>
    from .numbers import Rational
  File "/usr/local/lib/python3.9/site-packages/sympy/core/numbers.py", line 4567, in <module>
    _sympy_converter[type(mpmath.rational.mpq(1, 2))] = sympify_mpmath_mpq
AttributeError: module 'mpmath' has no attribute 'rational'
```

Information on the `mpmath` library installed with the cirq_pre_release image is as flollows.

```
Name: mpmath
Version: 1.4.0a0
Summary: Python library for arbitrary-precision floating-point arithmetic
Home-page:
Author:
Author-email: Fredrik Johansson <fredrik.johansson@gmail.com>
License: BSD
Location: /usr/local/lib/python3.9/site-packages
Requires:
Required-by: sympy
```

</details>

**Cirq version**
```
1.3.0
```

**Comment by kenya-sk** [comment]
Version 1.4.0 of mpmath is pre-release, so I will fix it to `mpmath~=1.3.0` once in the manual.

**Comment by NoureldinYosri** [comment]
@pavoljuhas as per offline discussions. I temporarily supressed the failing test to unblock development.
